kv/kvserver: TestRangefeedCheckpointsRecoverFromLeaseExpiration failed

kv/kvserver.TestRangefeedCheckpointsRecoverFromLeaseExpiration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11816592?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11816592?buildTab=artifacts#/) on master @ [6cbd07ee6fbfb92706e8cdc8c559960b1bc41663](https://github.com/cockroachdb/cockroach/commits/6cbd07ee6fbfb92706e8cdc8c559960b1bc41663):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:105
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:816
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:136
        	            	  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:816
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:715
        	            	  | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ClientInterceptor.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:228
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:900
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:785
        	            	  | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:908
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:211
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:189
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.raceTransport.SendNext
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport_race.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:2380
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1875
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1446
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1066
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnLockGatekeeper).SendLocked
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_lock_gatekeeper.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnMetricRecorder).SendLocked
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_metric_recorder.go:47
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).sendLockedWithRefreshAttempts
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:222
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).SendLocked
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:150
        	            	Wraps: (3) remote wall time is too far ahead (619.075229ms) to be trustworthy
        	            	Error types: (1) *markers.withMark (2) *errbase.opaqueWrapper (3) *errutil.leafError
        	Test:       	TestRangefeedCheckpointsRecoverFromLeaseExpiration
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3609167352
--- FAIL: TestRangefeedCheckpointsRecoverFromLeaseExpiration (8.52s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedCheckpointsRecoverFromLeaseExpiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31627

--- a/pkg/kv/kvserver/replica_rangefeed_test.go
+++ b/pkg/kv/kvserver/replica_rangefeed_test.go
@@ -1312,6 +1312,7 @@ func TestRangefeedCheckpointsRecoverFromLeaseExpiration(t *testing.T) {
 				WallClock: manualClock,
 			},
 			Store: &kvserver.StoreTestingKnobs{
+				DisableMaxOffsetCheck: true, // has been seen to cause flakes
 				TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
 					// Once reject is set, the test wants full control over the requests
 					// evaluating on the scratch range. On that range, we'll reject

